### PR TITLE
Blog app - integration specs for views

### PR DESCRIPTION
**Pull request by Somdotta07**
### Project requirements
-  Used Capybara to write integration tests for each view in your project.

- Login page:
  - I can see the username and password inputs and the "Submit" button.
  - When I click the submit button without filling in the username and the password, I get a detailed error.
  - When I click the submit button after filling in the username and the password with incorrect data, I get a detailed error.
  - When I click the submit button after filling in the username and the password with correct data, I am redirected to the root page.
  
- User index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.
  
- user show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.

- Post show page:
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commentor.
  - I can see the comment each commentor left.
Capture3

### CODE REVIEW

File: spec/requests/users_spec.rb [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
-  describe 'GET /show' do
-    user = User.create('name' => 'Amy', 'bio' => 'bio', 'photo' => 'Tom.png', 'posts_counter' => 0)
-    before(:each) { get user_path id: user.id }
+#   describe 'GET /show' do
+#     user = User.create('name' => 'Amy', 'bio' => 'bio', 'photo' => 'Tom.png', 'posts_counter' => 0)
+#     before(:each) { get user_path id: user.id }
 
-    it 'should return correct response' do
-      expect(response).to have_http_status(200)
-    end
```
Comment (Dipeshtwis) [inline spec/requests/users_spec.rb]: - [x] I would recommend removing all the commented code in different files if you are no longer using them.

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
+#     end
 
-    it 'checks if action rendered a correct template' do
-      expect(response).to render_template('index')
-    end
+#     it 'checks if action rendered a correct template' do
+#       expect(response).to render_template('index')
+#     end
 
-    it 'checks if correct placeholder is shown' do
```
Comment (Dipeshtwis) [inline spec/requests/posts_spec.rb]: - [x] I would recommend removing all the commented code in different files if you are no longer using them.

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -1,34 +1,34 @@
...
-    it 'check if the posts_counter numeric' do
-      subject.posts_counter = ''
-      expect(subject).to_not be_valid
-    end
+#     it 'check if the posts_counter numeric' do
+#       subject.posts_counter = ''
+#       expect(subject).to_not be_valid
+#     end
 
-    it 'check if the posts_counter greater than or equal to zero' do
```
Comment (Dipeshtwis) [inline spec/models/user_spec.rb]: - [x] I would recommend removing all the commented code in different files if you are no longer using them.

File: spec/models/post_spec.rb [hunk 4]
```diff
@@ -1,32 +1,30 @@
...
+#     end
 
-    it 'validates that likes counter is greater than or equal to 0' do
-      subject.likes_counter = -1
-      expect(subject).to_not be_valid
-    end
+#     it 'validates that likes counter is greater than or equal to 0' do
+#       subject.likes_counter = -1
+#       expect(subject).to_not be_valid
+#     end
```
Comment (Dipeshtwis) [inline spec/models/post_spec.rb]: - [x] I would recommend removing all the commented code in different files if you are no longer using them.

File: spec/models/like_spec.rb [hunk 5]
```diff
@@ -1,19 +1,19 @@
...
-  end
-end
+#   it 'updates a posts likes correctly' do
+#     user = User.new(name: 'Tom', posts_counter: 0)
+#     user2 = User.create!(name: 'Lilly', posts_counter: 0)
+#     post = user2.posts.create!(title: 'Post1', text: 'This is a post', likes_counter: 0, comments_counter: 0)
+#     post.likes.create!(user:)
+#     post.likes_counter = post.likes.length
+#     expect(post.likes_counter).to eql(1)
+#   end
```
Comment (Dipeshtwis) [inline spec/models/like_spec.rb]: - [x] I would recommend removing all the commented code in different files if you are no longer using them.

File: spec/models/comment_spec.rb [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
-    post.comments.create!(text: 'Hello World', user: comment_creator)
-    post.comments.create!(text: 'This is my second post', user: comment_creator)
-    expect(post.comments.length).to eql(2)
-  end
-end
+#     comment_creator = User.new(name: 'Jerry', photo: 'Tom.png', bio: 'bio', posts_counter: 0)
+#     post.comments.create!(text: 'Hello World', user: comment_creator)
+#     post.comments.create!(text: 'This is my second post', user: comment_creator)
+#     expect(post.comments.length).to eql(2)
+#   end
```
Comment (Dipeshtwis) [inline spec/models/comment_spec.rb]: - [x] I would recommend removing all the commented code in different files if you are no longer using them.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @Somdotta07, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N.A.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb (hunk 6 above)
Comment (Somdotta07) [reply spec/models/comment_spec.rb]: Thank you for your feedback. I'll fix it.

File: spec/models/user_spec.rb (hunk 3 above)
Comment (Somdotta07) [reply spec/models/user_spec.rb]: Thank you for your feedback. I'll fix it.

File: spec/models/post_spec.rb (hunk 4 above)
Comment (Somdotta07) [reply spec/models/post_spec.rb]: Thank you for your feedback. I'll fix it.

File: spec/models/like_spec.rb (hunk 5 above)
Comment (Somdotta07) [reply spec/models/like_spec.rb]: Thank you for your feedback. I'll fix it.

File: spec/requests/posts_spec.rb (hunk 2 above)
Comment (Somdotta07) [reply spec/requests/posts_spec.rb]: Thank you for your feedback. I'll fix it.

File: spec/requests/users_spec.rb (hunk 1 above)
Comment (Somdotta07) [reply spec/requests/users_spec.rb]: Thank you for your feedback. I'll fix it.

**Comment by Somdotta07** [comment]
Fixed issues suggested by the previous reviewer.

**Review by Dipeshtwis — APPROVED** [review]
Hi @Somdotta07, 

Great work on suggestions.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**